download-utils/src/lib.rs

```diff
@@ -329,3 +332,81 @@ pub fn download_snapshot_archive<'a, 'b>(
         desired_snapshot_hash.0, rpc_addr
     ))
 }
+
+/// Checks if the given RPC node has a newer incremental snapshot available
+/// Returns Some(slot, hash) if a newer slot is available
+/// Returns None otherwise
+pub fn check_for_newer_incremental_snapshot(
+    rpc_addr: SocketAddr,
+    full_snapshot_slot: u64,
+    incremental_snapshot_slot: u64,
+) -> Option<(u64, Hash)> {
+    let incremental_redirect_url =
+        &format!("http://{}{}", rpc_addr, INCREMENTAL_SNAPSHOT_REQUEST_PATH);
+    let response = reqwest::blocking::get(incremental_redirect_url);
+    let response = match response {
+        Ok(r) => r,
+        Err(_e) => {
+            return None;
+        }
+    };
+
+    let incremental_snapshot_url = String::from_str(response.url().path());
+    let incremental_snapshot_url = match incremental_snapshot_url {
+        Ok(url) => url,
+        Err(_e) => {
+            return None;
+        }
+    };
+
+    // Expected URL format: /incremental-snapshot-<full slot>-<incremental slot>-<incremental slot hash>.tar.zst
+    // eg: "/incremental-snapshot-152074294-152089382-BqBphVi1gnim96v6xXim7xq1L2PDFX3aDjd1Nn7SbRFM.tar.zst"
+    let re = Regex::new(r"incremental-snapshot-([0-9]*)-([0-9]*)-(\w*)\.").unwrap();
+    let captures = re.captures(&incremental_snapshot_url.as_str());
+    let captures = match captures {
+        Some(c) => c,
+        None => {
+            return None;
+        }
+    };
+    let full_slot = captures.get(1).map(|m| m.as_str().parse::<u64>());
+    let full_slot = match full_slot {
+        Some(s) => match s {
+            Ok(s) => s,
+            Err(_e) => {
+                return None;
+            }
+        },
+        None => {
+            return None;
+        }
+    };
```

Comment:
And this pattern can be something like:
```suggestion
    let full_slot = captures.get(1).and_then(|m| m.as_str().parse::<u64>().ok())?;
```
([and_then()](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then))